Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -390,20 +390,32 @@ def bottle_formula(f)
       mismatches = [:root_url, :prefix, :cellar, :rebuild].reject do |key|
         old_spec.send(key) == bottle.send(key)
       end
-      mismatches.delete(:cellar) if old_spec.cellar == :any && bottle.cellar == :any_skip_relocation
+      if old_spec.cellar == :any && bottle.cellar == :any_skip_relocation
+        mismatches.delete(:cellar)
+        bottle.cellar :any if OS.linux?
```

Comment:
We shouldn't scatter `if OS.linux?` around the codebase but instead use the `extend/` pattern. There's also no explanation on why this would be the case or why `:any_skip_relocation` should be used on Linux.

Reply:
I'd need to understand why this fix is necessary to agree.

Reply:
When the old bottle is `cellar :any`, and the new bottle is detected as `:any_skip_relocation`, then it's safe to mark the bottle as `cellar :any`.

Reply:
@iMichka Could you please pare this PR down to keeping just this one chunk?

Reply:
Why is the bottle.cellar :any needed rather than deleting the mismatch?

Reply:
```
      if old_spec.cellar == :any && bottle.cellar == :any_skip_relocation
        mismatches.delete(:cellar)
        bottle.cellar :any
      end
```
This section is for `--keep-old`. The previous value is `:any`. The current value is `:any_skip_relocation`. If the previous bottles are in fact `:any`, they will not work with `:any_skip_relocation. The current value must be changed to `:any`.

Reply:
Gotcha 👍🏻